compiler/noirc_errors/src/reporter.rs

```diff
@@ -17,6 +17,7 @@ pub struct CustomDiagnostic {
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum DiagnosticKind {
     Error,
+    Bug,
```

Comment:
Just saw your comment in slack on this. After finishing looking through the PR I'm fine with `bug` now since it seems to be unique in being a higher-severity warning. I'd also be alright with changing it to warning or error though as long as we stop compilation for the later.

Reply:
I'll leave it as a bug for better visibility